lib/v2/blocs/rates/mappers/rates_state_mapper.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:seeds/v2/datasource/remote/model/fiat_rate_model.dart';
+import 'package:seeds/v2/datasource/remote/model/rate_model.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/blocs/rates/viewmodels/rates_state.dart';
+
+class RatesStateMapper extends StateMapper {
+  RatesState mapResultToState(RatesState currentState, List<Result> results) {
+    if (areAllResultsError(results)) {
+      return currentState.copyWith(pageState: PageState.failure, errorMessage: 'Cannot fetch balance...');
```

Comment:
Since this is a sevice in the background I don't know if the user should be informed of the error with a snack or full screen globally, the only cases that comes to mind does not have internet or the end-point does not work.

Reply:
we need to have a global level catch for no internet and show the user when the app doesnt have internet

Reply:
but yeah, i dont think we need to display error when this call fails